sdk/core/scheduler/thread.h

```diff
@@ -616,7 +647,12 @@ namespace
 		uint8_t priority;
 		/// The original priority level for this thread.  This never changes.
 		const uint8_t OriginalPriority;
-		ThreadState   state;
+		ThreadState   state : 2;
+		/**
+		 * If the thread is yielding, it may be scheduled before its timeout
+		 * expires, as long as no other threads are runnable.
```

Comment:
I might leave off the "as long as..." part here and just point at the docs for `ThreadSleepNoEarlyWake`?

Reply:
I think it's fine to have duplicated explanations, especially in doc comments.  If you mouse over a use of this field in VS Code (for example), you'll get this doc comment pop up.  Having it tell you to go and read other docs can be annoying.

Reply:
That's fair.  In which case...
```suggestion
		 * expires, as long as no other threads are runnable or
		 * sleeping with shorter timeouts.
```